src/istgt.c

```diff
@@ -329,21 +329,28 @@ istgt_pg_match_all(PORTAL_GROUP *pgp, CF_SECTION *sp)
 			return 0;
 		rc = istgt_parse_portal(portal, &host, &port, &que);
 		if (rc < 0)
-			return 0;
+			goto no_match;
 		if (strcmp(pgp->portals[i]->label, label) != 0)
-			return 0;
+			goto no_match;
 		if (strcmp(pgp->portals[i]->host, host) != 0)
-			return 0;
+			goto no_match;
 		if (strcmp(pgp->portals[i]->port, port) != 0)
-			return 0;
+			goto no_match;
 		if (pgp->portals[i]->que != que)
-			return 0;
+			goto no_match;
 	}
 	label = istgt_get_nmval(sp, "Portal", i, 0);
 	portal = istgt_get_nmval(sp, "Portal", i, 1);
 	if (label != NULL || portal != NULL)
 		return 0;
 	return 1;
+
+no_match:
```

Comment:
port and host may be leaked if we return at line 345 and 346.

Reply:
ok